src/dcos_e2e/node.py

```diff
@@ -139,11 +140,46 @@ def _install_dcos_from_node_path(
             log_output_live: If ``True``, log output live.
             transport: The transport to use for communicating with nodes. If
                 ``None``, the ``Node``'s ``default_transport`` is used.
+            files_to_copy_to_genconf_dir: Pairs of host paths to paths on
+                the installer node. These are files to copy from the host to
+                the installer node before installing DC/OS.
+
         """
         tempdir = Path(gettempdir())
 
         remote_genconf_dir = 'genconf'
         remote_genconf_path = remote_build_artifact.parent / remote_genconf_dir
+
+        for host_path, installer_path in files_to_copy_to_genconf_dir:
+            # This is a hack that only works because backends are abusing
+            # the bug of being able to supply ``ip_detect_contents`` in the
+            # DC/OS configuration file ``config.yaml``.
+            # As long as a ``Node`` is not aware of its backend this is the
+            # only way to select the backend-specific ``ip-detect`` script.
+            if installer_path == Path('/genconf/ip-detect'):
+                try:
+                    del dcos_config['ip_detect_contents']
+                except KeyError:
+                    pass
```

Comment:
Will we hit this `pass` in coverage? (no need to answer, the report will tell us)